Dima/netflix-roulette/src/Pages/FilmPage/FilmPage.tsx

```diff
@@ -33,32 +29,33 @@ type PropsFromRouteAndRedux = ConnectedProps<typeof connector> & RouteComponentP
 
 
 class FilmPage extends React.Component<PropsFromRouteAndRedux> {
-  requestsChain = async() => {
-    await this.props.CurrentFilmRequested(this.props.match.params.id) 
-    const sortBy = 'vote_average'
-    const searchBy = 'genre'
-    const search = this.props.currentFilm.genres[0]
-    await this.props.FilmsRequested(sortBy, searchBy, search)
-    await console.log(sortBy, searchBy, search)
-  }
 
   componentDidMount() {
-    this.requestsChain()
+    this.props.currentFilmRequested(this.props.match.params.id) 
   };
   
   componentDidUpdate(prevProps: PropsFromRouteAndRedux) {
+    if(this.props.currentFilm !== prevProps.currentFilm) {
+      const search = this.props.currentFilm.genres[0]
+      console.log(search)
+      this.props.filmsRequested(0, 'vote_average', 'genre', search)
+    }
+    if(this.props.match.params.id !== prevProps.match.params.id) {
+      this.props.currentFilmRequested(this.props.match.params.id) 
+    }
+  }
+
+  handlePagination = (offset: number, pagination: boolean = false) => {
     const URLData = parse(this.props.location.search) as { 
-      sortBy: string,
-      searchBy: string,
-      search: string
+      sortBy: string, 
+      searchBy: string, 
+      search: string 
     };
-    const { sortBy } = URLData
-    const searchBy = 'genre'
-    const search = this.props.currentFilm.genres[0]
-    this.props.match.params.id !== prevProps.match.params.id 
-      && this.props.CurrentFilmRequested(this.props.match.params.id) 
-        && this.props.FilmsRequested(sortBy, searchBy, search)
+    const { sortBy, searchBy, search } = URLData 
+    this.props.filmsRequested(offset, sortBy, searchBy, search, pagination)
```

Comment:
on Film Page you have to load more movies with SAME genre. You have to you default sortings and genre that you take from movie. And here you just get params from url.